src/librustc_trans/trans/intrinsic.rs

```diff
@@ -589,217 +582,115 @@ pub fn trans_intrinsic_call<'a, 'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
+                    ast::TyI8 => 8,
+                    ast::TyI16 => 16,
+                    ast::TyI32 => 32,
+                    ast::TyI64 => 64,
+                };
+                match c {
+                    "ctlz" => count_zeros_intrinsic(bcx, &format!("llvm.ctlz.i{}", width),
+                                                    llargs[0], call_debug_location),
+                    "cttz" => count_zeros_intrinsic(bcx, &format!("llvm.cttz.i{}", width),
+                                                    llargs[0], call_debug_location),
+                    "ctpop" => ccx.get_intrinsic(&format!("llvm.ctpop.i{}", width)),
+                    "bswap" => {
+                        if width == 8 {
+                            llargs[0] // byte swap a u8/i8 is just a no-op
+                        } else {
+                            ccx.get_intrinsic(&format!("llvm.bswap.i{}", width))
+                        }
+                    }
+                    "add_with_overflow" =>
+                        with_overflow_intrinsic(
+                            bcx, &format!("llvm.sadd.with.overflow.i{}", width),
+                            llargs[0], llargs[1], llresult, call_debug_location),
+                    "sub_with_overflow" =>
+                        with_overflow_intrinsic(
+                            bcx, &format!("llvm.ssub.with.overflow.i{}", width),
+                            llargs[0], llargs[1], llresult, call_debug_location),
+                    "mul_with_overflow" =>
+                        with_overflow_intrinsic(
+                            bcx, &format!("llvm.smul.with.overflow.i{}", width),
+                            llargs[0], llargs[1], llresult, call_debug_location),
+                    "overflowing_add" => Add(bcx, llargs[0], llargs[1], call_debug_location),
+                    "overflowing_sub" => Sub(bcx, llargs[0], llargs[1], call_debug_location),
+                    "overflowing_mul" => Mul(bcx, llargs[0], llargs[1], call_debug_location),
+                    "unchecked_div" => SDiv(bcx, llargs[0], llargs[1], call_debug_location),
+                    "unchecked_rem" => SRem(bcx, llargs[0], llargs[1], call_debug_location),
+                    _ => unreachable!(),
+                }
+            } else if let &TyUint(t) = sty {
+                let width = match t {
+                    ast::TyUs => {
+                        match &ccx.tcx().sess.target.target.target_pointer_width[..] {
+                            "32" => 32,
+                            "64" => 64,
+                            tws => panic!("Unsupported target word size for usize: {}", tws),
+                        }
+                    },
+                    ast::TyU8 => 8,
+                    ast::TyU16 => 16,
+                    ast::TyU32 => 32,
+                    ast::TyU64 => 64,
+                };
+                match c {
+                    "ctlz" => count_zeros_intrinsic(bcx, &format!("llvm.ctlz.i{}", width),
```

Comment:
Unfortunately no, as they are two different types. The most I could do is unduplicate the target_pointer_width part.